### ISSUE TITLE
Exception on constant checkout using `DB:Pool` with `HTTP::Client`

### ISSUE THREAD

**Issue by syeopite** [crystal-lang/crystal-db#210]
Server
```crystal
require "http/server"

server = HTTP::Server.new do | context |
    context.response.content_type = "plain/text"
    context.response.print "content"
end

address = server.bind_tcp 8080
server.listen
```

Client

```crystal
pool = DB::Pool(HTTP::Client).new() {HTTP::Client.new("127.0.0.1", 8080)}
  
loop do
  pool.checkout &.get("/") do | r |
  rescue ex
    p! ex.class, ex.message
  end
end
```

Produces

```
Unhandled exception: Unsupported HTTP version: contentHTTP/1.1 (Exception)
  from /usr/lib/crystal/http/client/response.cr:134:13 in '__crystal_main'
  from /usr/lib/libc.so.6 in '??'
  from /usr/lib/libc.so.6 in '__libc_start_main'
  from ./app in '_start'
  from ???
```

Where it seems like the response and headers are merged together for some reason.

In an actual application when the first exception doesn't terminate the program the factory block of `DB::Pool` is also never called again on subsequent checkouts

**Comment by bcardiff** [comment]
It ended up working what you were trying to do @syeopite ?

**Comment by syeopite** [comment]
Partially. I fixed the issue where the response and headers were merged together so I figured it may just be something I'm doing wrong rather than a bug upstream 

But the main problem I was facing and not really shown in the reproducible example I gave above was with HTTP::Clients constructed with an IO 

Afaik since these clients aren't able to reconnect the dead idling clients needs to be explicitly deleted from the pool with the #delete method

What I can't figure out is why even with new clients made from the factory block, the clients still aren't able to connect to the server. 

Here's something that's closer to my actual code https://bpa.st/JXXQ

If you modify the server example I gave above to track the total number of requests the number of requests goes up fine.

But as soon as you terminate the binary and then relaunch it, even new clients from the pool are forever unable to connect resulting in a constant total request of zero on the server side.
